data/src/main/java/org/apache/iceberg/data/orc/GenericOrcReader.java

```diff
@@ -351,20 +441,43 @@ private static Converter buildConverter(final Types.NestedField icebergField,
       case SHORT:
         return new ShortConverter();
       case DATE:
+        return new DateConverter();
       case INT:
         return new IntConverter();
       case LONG:
-        return new LongConverter();
+        ORCSchemaUtil.LongType longType =
+            ORCSchemaUtil.LongType.valueOf(schema.getAttributeValue(ORCSchemaUtil.ICEBERG_LONG_TYPE_ATTRIBUTE));
```

Comment:
Is the spec up to date with these ORC metadata fields?

Reply:
Updated the spec now